### ISSUE TITLE
Advanced search doesn't return any result if search by custom fields

### ISSUE THREAD

**Issue by romannazarenko** [snipe/snipe-it#5890]
#### Please confirm you have done the following before posting your bug report:

- [yes ] I have enabled debug mode 
- [ yes] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

**Describe the bug**
I have couple of custom fields in assets. I try to advanced search by these fields, but search return empty results. If i search by predefined fields (Asset Tag, Serial, Model, Category, Status, Checked Out To) search working fine. 

**To Reproduce**
1. Go to 'Assets - List all'
2. Click on 'Advanced Search icon'
3. Enter search terms in any of custom fields
4. Search return 0 results

**Expected behavior**
Search return all assets with searched data in custom field

**Screenshots**
![image](https://user-images.githubusercontent.com/25567563/42930293-415eeb00-8b45-11e8-8df2-c69a0b3b4627.png)

**Server (please complete the following information):**
 - Snipe-IT Version v4.6.0 - build 3710 (master)
 - OS: [e.g. Ubuntu, CentOS] centOS7
 - Web Server: [e.g. Apache, IIS] Apache
 - PHP Version 7.1.18

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] Windows 10
 - Browser [e.g. chrome, safari] Chrome
 - Version [e.g. 22] 67.0.3396.99 (Official Build) (64-bit)

**Error Messages**
- Confirm whether the error is reproducible on the demo: yes, same behavior in Demo

**Additional context**
- Is this a fresh install or an upgrade? Trying on fresh install and upgraded from 4.5, 4.4
- What method you used to install Snipe-IT (install.sh)

**Comment by tilldeeke** [comment]
I can reproduce this, but couldn't find the reason for the bug so far. The generated api request is correct and returns the correct assets, but the client side javascript somehow doesn't display the result correctly. 

I'm going to look deeper into this and I'll update you when I find something. 😊

**Comment by tilldeeke** [comment]
Alright, from what I can tell, this seems to be a bug in the bootstrap tables libraries.

The filtering in the toolbar sets the `filterColumnsPartial` property:
https://github.com/snipe/snipe-it/blob/1956a16d1e3b8f507a3b0bd4d30918e25fec1df6/public/js/extensions/toolbar/bootstrap-table-toolbar.js#L200-L204
Since the property is not empty, the `getData` method tries to return `this.data`. After the search, the results are saved in `this.options.data`, but not in `this.data` though. I stepped throught this funtion with the debug tools in Chrome, but only got data in `this.options.data`.
https://github.com/snipe/snipe-it/blob/1956a16d1e3b8f507a3b0bd4d30918e25fec1df6/public/js/bootstrap-table.js#L2376-L2380

I'm sorry, but this is as far as I got, maybe someone else can figure out how to solve this issue.
